ASM - Add documentation for custom WAF rules (#21218)
DataDog/documentation: content/en/security/application_security/threats/add-user-info.md

{{< /programming-lang-wrapper >}}

### Tracking business logic information without modifying the code

> If your service has ASM enabled and [Remote Config configured][1], you may create a custom WAF rule that'll flag any request it matches with a custom business logic tag. This doesn't require any modification to your application, and can be done entirely from Datadog.

Reviewer: ```suggestion
If your service has ASM enabled and [Remote Configuraton][1] enabled, you can create a custom WAF rule to flag any request it matches with a custom business logic tag. This doesn't require any modification to your application, and can be done entirely from Datadog.
```

To get started, navigate to the [Custom WAF Rule page][2] and click on "Create New Rule".

{{< img src="security/application_security/threats/custom-waf-rule-menu.png" alt="Access the Custom WAF Rule Menu from the ASM homepage by clicking on Protection, then In-App WAF and Custom Rules" style="width:100%;" >}}

> This will open a menu in which you may define your custom WAF rule. By selecting the "Business Logic" category, you will be able to configure an event type (for instance, `users.password_reset`). You can then select the service you want to track, and a specific endpoint. You may also use the rule condition to target a specific parameter to identify the codeflow you want to "instrument". When the condition match, the library will tag the trace and flag it to be forwarded to ASM. If you don't need the condition, you may set a very broad condition to match everything.

Reviewer: ```suggestion
This will open a menu in which you may define your custom WAF rule. By selecting the "Business Logic" category, you will be able to configure an event type (for instance, `users.password_reset`). You can then select the service you want to track, and a specific endpoint. You may also use the rule condition to target a specific parameter to identify the codeflow you want to _instrument_. When the condition matches, the library tags the trace and flags it to be forwarded to ASM. If you don't need the condition, you may set a broad condition to match everything.
```

{{< img src="security/application_security/threats/custom-waf-rule-form.png" alt="Screenshot of the form that appear when you click on the Create New Rule button" style="width:50%;" >}}

> Once saved, the rule will be deployed to instances of the service with Remote Config enabled.

Reviewer: ```suggestion
Once saved, the rule is deployed to instances of the service that have Remote Configuration enabled.
```
